formPage/my-app/src/ExcusedAbsenceForm.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+
+async function getCheckin(state) {
+    const request = await fetch("http://localhost:5000/submitExcusedAbsence", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+            "name": state.name,
+            "type": state.type,
+            "reason": state.reason
+        }),
+      });
+      // Get Status and Response
+      const resp = await request.json();
+      const msg = await resp.msg;
+      return msg;
+        
+}
+
+
+export default class ExcusedAbsenceForm extends React.Component {
```

Comment:
Let's make a decision to use class-based or function-based components throughout the entire project. Is there a reason why we use class based here?